### ISSUE TITLE
[SUGGESTION] Add the missing launcher versions

### ISSUE THREAD

**Issue by VesuviusVenox** [PolyMC/PolyMC#481]
### Role

I play older Minecraft versions quite often. 

### Suggestion

Have all the currently found alpha, infdev, indev and classic versions available for direct download through the launcher. 

### Benefit

To give people the option to experience old Minecraft with more ease.

### This suggestion is unique

- [X] I have searched the issue tracker and did not find an issue describing my suggestion, especially not one that has been rejected.

### You may use the editor below to elaborate further.

The launcher is severely lacking in a lot of versions which have since been discovered, ranging from alpha to early classic. Betacraft has all currently found versions available from the get-go, so it would be nice to have this in PolyMC as well. 
Full archive of missing versions: 
Classic: https://archive.org/download/Minecraft-JE-Classic
Indev: https://archive.org/download/Minecraft-JE-Indev
Infdev: https://archive.org/download/Minecraft-JE-Infdev
Alpha: https://archive.org/download/Minecraft-JE-Alpha
Beta: https://archive.org/download/Minecraft-JE-Beta

**Comment by glowiak** [comment]
It's good, but the launcher needs a json file with versions, I see there are no jsons

**Comment by TheMCNerd2017** [comment]
This would be an amazing addition to the launcher. However there are a few small potential issues that would have to be addressed.

1. Like @glowiak  stated there is no JSON file with the versions present (a version index) in any of those archives. One would have to be created and maintained for the launcher to have the ability to see and download them.
2. The Internet Archive can be slow and goes under maintenance sometimes which can make files temporarily inaccessible. Users might not know this and be confused when they get an error when attempting to download a version from one of the archives through the launcher while IA is down.
3. Some of those versions there are still provided by Mojang/Microsoft (or have the exact same names), so the launcher would need to differentiate between versions from Mojang/Microsoft and versions from those archives.
4. There is a very slim chance that Mojang/Microsoft could decide one day to file DMCA claims against those archives. If that happens this functionality would become broken if the files aren't mirrored elsewhere.

**Comment by HalJordan** [comment]
On point 2(and possibly 4) couldn't you just run a check to see if they're down for whatever reason, then show an appropriate message on the install screen as to why those aren't available?

**Comment by TheMCNerd2017** [comment]
> On point 2(and possibly 4) couldn't you just run a check to see if they're down for whatever reason, then show an appropriate message on the install screen as to why those aren't available?

Yeah that would be a good solution for if the files are unavailable or down for any reason. You'd have to parse the error from IA though since different ones can come up. For the potential slowness a message could be shown notifying users of that as well. Maybe upon first downloading a version from those archives. A workaround to the slowness is by using multiple connections during download, however seeing as IA provides all this stuff for free and runs on donations it isn't really fair to IA.

Also another point I should have added is that IA can require a valid login to download certain items, but so far they only have that requirement for very popular items that take up a lot of their bandwidth like ROM sets.

**Comment by HalJordan** [comment]
It should only be checking this when you have that page open, so it's not like we should be hitting it any time PolyMC is just sitting there idle. So I don't see this really impacting them any more than usual. But it may be necessary to put in the authentication part in case they ever lock down the old MC versions if they become popular enough. An ounce of prevention and all.

**Comment by LennyMcLennington** [comment]
IA is also blocked by default on some ISPs since its purpose is archival it doesn't tend to censor stuff that's not appropriate for children.
